README.md

```diff
@@ -30,165 +32,555 @@ The two guiding principles are:
+_Note: If you prefer to use `yarn`, you can substitute the above command with `yarn create hops-app my-hops-app`._
+
+Then move into the newly created directory:
 
 ```shell
-$ yarn create hops-app@next --hops-version next --template hops-template-react@next my-awesome-project
-# OR npx create-hops-app@next --hops-version next --template hops-template-react@next my-awesome-project
-$ cd my-awesome-project
-$ yarn start
+cd my-hops-app
 ```
 
-This will start hops in development mode. Visit [http://localhost:8080](http://localhost:8080) to see your app in the browser and make some changes to the code in your editor to see it live-reloading.
+And start the development server:
 
-### Templates
+```shell
+npm start
+```
 
-Hops provides a few templates as starting points, which can be installed using the `--template` option of the [Hops CLI `init` command](https://github.com/xing/hops/tree/master/packages/cli).
+This will start Hops in development mode. Visit [http://localhost:8080](http://localhost:8080) to see your app in the browser and make some changes to the code in your editor to see it live-reloading.
 
-#### [`hops-template-react`](https://github.com/xing/hops/tree/master/packages/template-react) (default)
+## Walk-through
 
-This is the default template, it consists of a simple React starting point and it is being used when you execute:
+### Entry files and `hops`'s `render()` function
 
-```bash
-$ hops init --hops-version next --template hops-template-react@next my-awesome-project
+Hops assumes your entry file is either named `index.js` or is specified via the [`"main"` field in your `package.json`](https://docs.npmjs.com/files/package.json#main). This is similar to how Node.js would resolve a package.
+
+**`package.json`**
+
+```json
+{
+  "main": "./src/app.js"
+}
 ```
 
-#### [`hops-template-redux`](https://github.com/xing/hops/tree/master/packages/template-redux)
+The main export of the entry file must be the result of `hops`'s [`render()`](#render) function:
 
-In case you want to build a Redux project, you can use this template as a starting point.
+**`src/app.js`**
+
+```javascript
+import { render } from 'hops';
+import React from 'react';
 
-```bash
-$ hops init --hops-version next --template hops-template-redux@next my-awesome-project
+export default render(<h1>Hello World!</h1>);
 ```
 
-#### [`hops-template-graphql`](https://github.com/xing/hops/tree/master/packages/template-graphql)
+### Routing and `hops`'s `<Miss />` component
 
-In case you want to build a GraphQL project, you can use this template as a starting point.
+Hops ships with [`react-router`](https://reacttraining.com/react-router/web/guides/quick-start) in order to render different components per route. Additionally Hops provides some helper components, such as `<Miss />` which will tell the server that no matching route has been found:
```

Comment:
It does not really instruct the server to answer with a `404`, but just tells it that our rendering middleware is not responsible for the current request. If another middleware is registered after ours, it can answer in any way. If there is none, Express does answer using a generic fallback handler that sends the 404 we are usually seeing.

We could probably say something along these lines, though: `...which is equivalent to calling 'next()' in an Express middleware...`.

Reply:
`(this usually means that Express.js answers with a 404 status code, if no other middleware is registered that handles this request)`